### DISCUSSION TITLE
MIUI autostart option

### DISCUSSION THREAD

**Discussion by barteqcz** [d4rken-org/sdmaid-se#550] [Ideas]
I have a suggestion for SD Maid to use autostart in MIUI. The point is that if we do not enable the autostart option for the SD Maid application, then each time the application is closed, the ability to enable AppCleaner will be broken. That's because the accessibility service turns off then, causing some weird bug, and you have to turn it back on, which is incredibly inconvenient if it's there every time you restart the app. 

A potential solution would be to either enable autostart by default after application installation if MIUI is detected, or add information about the need to enable this feature in the list of required permissions, shown after the application installation.

Cheers!

**Comment by d4rken** [comment]
I'm not sure what MIUI means with "autostart" is it the inclusion of an ON_BOOT receiver?

**Comment by barteqcz** [comment]
it's something what's fully called 'background autostart' and it allows apps to start in the background automatically. However, I don't know about details of this feature, but many apps use this by default, like Messenger, Google Photos, and many other apps.

[Screenshot_2023-08-01-13-41-55-356_com miui securitycenter-edit](https://github.com/d4rken-org/sdmaid-se/assets/118764028/131985a5-e89f-40f2-bcde-edc838eaf58c)
[Screenshot_2023-08-01-13-44-15-654_com miui securitycenter](https://github.com/d4rken-org/sdmaid-se/assets/118764028/2e20c581-eadf-4d4a-a8a2-699a4774fba4)

**Reply by d4rken** [reply]
I don't know how to "turn on" this setting by default. I have the suspicion that Google apps are just enabled by default.

**Reply by barteqcz** [reply]
I don't think so, honestly. Many apps that I have from Google Play, aren't enabled by default. Many Google apps aren't enabled too. Honestly, even a simple reminder in the SD Maid SE permission list that appears after the installation would be good, just to inform the user. Even a simple thing like this:

"Since you're running MIUI, you have to enable autostart permission for SD Maid SE, to prevent the accesibility service from disabling after closing the app. You can do that in Settings>Apps>Permissions>Background autostart by enabling SD Maid on the list."

would be good (sorry for my English, though). But I don't know, whether it is possible to detect, if the user runs MIUI.

**Reply by barteqcz** [reply]
Actually, now I see that such a description is there already and it actually responds to this. Sorry; if it was there earlier, I just didn't see it.

**Reply by d4rken** [reply]
It has been there the whole time. But it's a lot of text. Users don't like reading text 😉. Not sure how I can make this easier. Would love to automate the setup here, but have not found a way.
Any ideas?

**Reply by barteqcz** [reply]
Good question. You're 100% right, that it's better to avoid putting such a long text there 😄. It would be nice if there was any guide on how to apply that autostart by default, but that's still a good question too.

**Reply by barteqcz** [reply]
I found something, that might help you!
https://dontkillmyapp.com/xiaomi
https://github.com/XomaDev/MIUI-autostart
![image](https://github.com/d4rken-org/sdmaid-se/assets/118764028/bb3fc08b-2b8a-4716-be86-b19106d2ef30)

**Reply by d4rken** [reply]
That's very interesting. I could add a specific setup card for this, :thinking: 

Not sure if using `org.lsposed.hiddenapibypass` will get me intro trouble with big G :thinking:.

**Reply by barteqcz** [reply]
Yeah, honestly - good question.
